### PR TITLE
SNOW-3190420: Add start_nbctl.sh validation check for Notebook custom…

### PR DESCRIPTION
**Pull request by sfc-gh-jijiang**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.
   * [ ] I've described my changes in the documentation.

### Changes description

> **Note:** This PR is stacked on top of https://github.com/snowflakedb/snowflake-cli/pull/2809(custom image validation framework) which is still under review. Please treat this as an early preview — it is ready for feedback but should not be merged until the parent PR lands.

**SNOW-3190420: Add `start_nbctl.sh` validation check for Notebook custom images**

This PR adds a Notebook-specific validation check to the `snow custom-image validate` CLI command.

For Notebook vNext, PEP injects `./start_nbctl.sh /shared/nbctl` as the container entrypoint args. If this script is missing or not executable in the custom image, the container will crash immediately on startup. This check catches that before deployment.

**What changed:**
- **`image_validation.yaml`**: Added `required_scripts: [start_nbctl.sh]` config entry
- **`manager.py`**: Added `_check_required_scripts` handler that runs `docker run ... test -x <script>` to verify each script exists and is executable
- **`test_manager.py`**: Added unit tests for pass (script present) and fail (script missing) scenarios, updated mock helper to support `missing_scripts` parameter

**Validation:**
- All 19 unit tests pass (`python3 -m pytest tests/custom_images/ -v`)
- Manually verified `start_nbctl.sh` exists and is executable in the official `snowflake-notebook:ml-cpu` image
- Run the script locally and verified the start_nbctl part works:
image
image

### CODE REVIEW

File: tests/custom_images/test_manager.py [hunk 1]
```diff
@@ -223,6 +225,61 @@ def test_checks_skipped_when_not_configured(self, mock_run, tmp_path):
         assert "dependency_health" not in check_names
         assert "vulnerability_scan" not in check_names
 
+    @mock.patch("snowflake.cli._plugins.custom_images.manager.subprocess.run")
+    def test_required_scripts_pass(self, mock_run, config_path):
```
Comment (sfc-gh-olorek) [inline tests/custom_images/test_manager.py]: Please also add a test on required_scripts being an empty list

File: tests/custom_images/test_manager.py [hunk 2]
```diff
@@ -72,6 +75,12 @@ def side_effect(*args, **kwargs):
                         stdout=pip_check_result[1],
                         stderr="",
                     )
+                elif "test" in cmd and ("-x" in cmd or "-f" in cmd):
+                    target = cmd[-1]
+                    failed = any(s in target for s in missing_scripts)
+                    return mock.Mock(
+                        returncode=1 if failed else 0, stdout="", stderr=""
```
[the remark this answers is not in the document]
Comment (sfc-gh-olorek) [reply tests/custom_images/test_manager.py]: @sfc-gh-jijiang please address

File: src/snowflake/cli/_plugins/custom_images/manager.py [hunk 3]
```diff
@@ -290,6 +291,45 @@ def _check_entrypoint(
             message=f"Entrypoint is correctly set to '{expected}'",
         )
 
+    def _check_required_scripts(
```
Comment (sfc-gh-olorek) [inline src/snowflake/cli/_plugins/custom_images/manager.py:323]: This check is Notebook-specific but runs for all custom-image validate invocations. Users validating ML Job images will
  fail on a script they don't need. Consider gating this behind a --workload-type flag or splitting configs per workload.
